### PR TITLE
WIP: Multiple attachment support

### PR DESCRIPTION
**Pull request by Eldenroot**
WIP

- editpost.php
- newreply.php
- newthread.php
- template changes - post_attachments_new

Link to original thread: https://community.mybb.com/thread-216548.html; @azalea4va

### CODE REVIEW

**Comment by Eldenroot** [comment]
@Shade- - please, post me the link for these "coding standards" which are used by MyBB, thank you

File: editpost.php [hunk 1]
```diff
@@ -196,16 +196,36 @@
...
+		{
+			// Already in multi-attachment array format
+			$attachments = $_FILES;
+		}
+		else
+		{
+			// Convert original-style non-array $_FILES['attachment'][$field] to array format in $attachments
+			$attachments = array('attachment' => array());
+		foreach($fields as $field)
+			$attachments['attachment'][$field] = array($_FILES['attachment'][$field]);
```
Comment (effone) [inline editpost.php]: Where is the open brace?

File: editpost.php [hunk 2]
```diff
@@ -196,16 +196,36 @@
...
+		}
+		foreach($attachments['attachment']['name'] as $key => $name)
+		{
+			// Convert array $attachments['attachment'][$field][$key] to non-array format in $FILE
+			$FILE = array('attachment' => array());                
+		foreach($fields as $field)
+			$FILE['attachment'][$field] = $attachments['attachment'][$field][$key];
```
Comment (effone) [inline editpost.php]: Not good. Puzzled open brace ...

File: editpost.php [hunk 3]
```diff
@@ -216,6 +236,7 @@
 	{
 		$mybb->input['action'] = "editpost";
 	}
+		}
```
Comment (effone) [inline editpost.php]: Indent? Okay?

**Comment by Sama34** [comment]
Are you going to update this Eldenroot ?

**Comment by Eldenroot** [comment]
I will try during this weekend :) If there are more requested changes, let
me know to fix them all at once.

>

File: install/resources/mybb_theme.xml [hunk 4]
```diff
@@ -9475,9 +9475,9 @@ if(use_xmlhttprequest == "1")
 <td class="trow_shaded" style="white-space: nowrap"><em><a href="attachment.php?aid={$attachment['aid']}" target="_blank">{$attachment['filename']}</a> ({$attachment['size']})</em></td>
 <td class="trow_shaded" style="white-space: nowrap; text-align: center;">{$attach_mod_options} <input type="submit" class="button" name="rem" value="{$lang->remove_attachment}" onclick="return Post.removeAttachment({$attachment['aid']});" /> {$postinsert}</td>
 </tr>]]></template>
-		<template name="post_attachments_new" version="1807"><![CDATA[<tr>
+		<template name="post_attachments_new" version="1816"><![CDATA[<tr>
 <td class="trow1" width="1"><img src="{$theme['imgdir']}/paperclip.png" alt="" /></td>
-<td class="trow1" style="white-space: nowrap"><strong>{$lang->new_attachment}</strong> <input type="file" name="attachment" size="30" class="fileupload" /></td>
+<td class="trow1" style="white-space: nowrap"><strong>{$lang->new_attachment}</strong> <input type="file" name="attachment[]" size="30" class="fileupload" multiple/></td>
```
Comment (Sama34) [inline install/resources/mybb_theme.xml]: Should use multiple="multiple"

**Comment by Eldenroot** [comment]
There will be a new PR after I clean up my fork, all my opened PRs here messed up my branches in my local repo. I am moving to VScode sw...

**Pull request closed without merging**